packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -109,76 +54,430 @@ const invalidateCustomerBasketsQuery = (
 ): CacheUpdate => {
     if (!customerId) return {}
     return {
-        invalidate: [
-            and(
-                matchParameters(toGetCustomerBasketsParameters(customerId, parameters)),
-                matchesPath(getCustomerBasketsPath({...parameters, customerId}))
-            )
-        ]
+        invalidate: [{queryKey: getCustomerBaskets.queryKey({...parameters, customerId})}]
     }
 }
 
-const updateBasket = (
-    customerId: string | null,
-    {parameters}: {parameters: BasketParameters},
-    response: Basket
-): CacheUpdate => ({
-    // TODO: We only update the basket from the matching locale; we should also invalidate other locales
-    ...updateBasketQuery(customerId, parameters, response)
-})
-
-const updateBasketWithResponseBasketId = (
-    customerId: string | null,
-    {parameters}: {parameters: Omit<BasketParameters, 'basketId'>},
-    response: Basket
-): CacheUpdate => {
-    const {basketId} = response
-    return {
-        // TODO: We only update the basket from the matching locale; we should also invalidate other locales
-        ...(basketId && updateBasketQuery(customerId, {...parameters, basketId}, response))
-    }
-}
-
-/** Logs a warning to console (on startup) and returns nothing (method is unimplemented). */
-const TODO = (method: keyof Client) => {
-    console.warn(`Cache logic for '${method}' is not yet implemented.`)
-    return undefined
-}
-
 export const cacheUpdateMatrix: CacheUpdateMatrix<Client> = {
-    addCouponToBasket: updateBasket,
-    addGiftCertificateItemToBasket: TODO('addGiftCertificateItemToBasket'),
-    addItemToBasket: updateBasket,
-    addPaymentInstrumentToBasket: updateBasket,
-    addPriceBooksToBasket: TODO('addPriceBooksToBasket'),
-    addTaxesForBasket: TODO('addTaxesForBasket'),
-    addTaxesForBasketItem: TODO('addTaxesForBasketItem'),
-    createBasket: updateBasketWithResponseBasketId,
-    createShipmentForBasket: TODO('createShipmentForBasket'),
-    deleteBasket: (customerId, {parameters}) => ({
+    addCouponToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
```

Comment:
So we only update the customer/customerId/baskets cache, and not the basket/basketId anymore?

Reply:
The key that updates the basked cache is defined in the line above:
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/887efe78b951cb7b0a98d5168f621e718f77d0a3/packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts#L65